source/custom/nidcpower_service.custom.cpp

```diff
@@ -55,4 +55,18 @@ ::grpc::Status NiDCPowerService::MeasureMultiple(::grpc::ServerContext* context,
     }
   }
 
+::grpc::Status NiDCPowerService::ConvertApiErrorStatusForViSession(google::protobuf::int32 status, ViSession vi)
+{
+    const ViInt32 buffer_size = 4096;
+    ViStatus error_code {};
+    std::string description(buffer_size, '\0');
+    // Try first to get the most recent error with a dynamic message.
+    library_->GetError(vi, &error_code, buffer_size, description.data());
+    if (error_code != status) {
+        // Since another thread has changed the status, fall back to the static message lookup.
+        library_->ErrorMessage(vi, status, description.data());
```

Comment:
Bug: Since `trim_trailing_nulls` (used in `ApiErrorDescriptionToStatus`) only "erase[s] from the first non-null character (going backwards) to the end," you can run into scenarios like this:
* Line 64 fills in `description` with 47 characters
* Line 67 overwrites `description` with only 33 characters
* As a result, after the `'\0'` at character 34, characters 35-47 will still have the original data from line 64, and `trim_trailing_nulls` will only trim the string down to 47 characters.

You need to either update `trim_trailing_nulls` accordingly, or null out the description buffer again before line 67.

Reply:
Good point, I'll make sure to clear the buffer in each case before the second read.